### PR TITLE
Ability to take multiple measures on the same page

### PR DESCRIPTION
**Pull request by aomarks**
The `measurement` property in the JSON config file can now be an array, in which case all of those measurements are taken for each page load, and behave like their own benchmark, labeled as `benchmark [sub-benchmark]`.

For example:

```json
{
    "benchmarks": [
        {
            "name": "multi",
            "url": "multiple.html",
            "measurement": [
                {
                    "mode": "performance",
                    "entryName": "foo"
                },
                {
                    "mode": "performance",
                    "entryName": "bar"
                }
            ],
            "browser": {
                "name": "chrome",
                "headless": true
            }
        }
    ]
}
```

Screen Shot 2020-07-22 at 1 27 36 PM

Since each sub-measurement is treated as its own benchmark, we'll generate an NxN matrix as usual, which means each sub-benchmark is compared both across pages, and to other sub-benchmarks from the same page. I think we'll often not want to compare two sub-benchmarks from the same page, since they're measuring different things. But I think it could sometimes be what we want, so rather than treating sub-benchmarks as a special concept, I'm planning follow up work to add the ability to configure which measurements are compared to which in a general way (related to https://github.com/Polymer/tachometer/issues/47).

Fixes https://github.com/Polymer/tachometer/issues/34

### CODE REVIEW

**Comment by aomarks** [comment]
cc @andrewiggins FYI

File: src/automatic.ts [hunk 1]
```diff
@@ -228,44 +254,40 @@ export class AutomaticMode {
           `in ${spec.browser.name} from ${url}. Retrying.`);
     }
 
-    if (millis === undefined) {
+    if (numPending > 0) {
       console.log();
       throw new Error(
           `\n\nFailed ${maxAttempts}/${maxAttempts} times ` +
           `to get a measurement ` +
```
Comment (rictic) [inline src/automatic.ts:261]: In the case of multiple measurements, it'd be useful to know which measurement(s) are missing

File: src/automatic.ts [hunk 2]
```diff
@@ -192,19 +206,31 @@ export class AutomaticMode {
...
+      numPending = measurements.length;
       await openAndSwitchToNewTab(driver, spec.browser);
       await driver.get(url);
-      for (let waited = 0; millis === undefined && waited <= 10000;
-           waited += 50) {
+      for (let waited = 0; numPending > 0 && waited <= 10000; waited += 50) {
         // TODO(aomarks) You don't have to wait in callback mode!
         await wait(50);
-        millis = await measure(driver, spec.measurement, server);
+        for (let i = 0; i < measurements.length; i++) {
```
Comment (justinfagnani) [inline src/automatic.ts:224]: With this level of nested loops and conditionals, I'm finding it a bit hard to follow. Some more comments would be nice and probably some more descriptive variable names. Renaming `i` to something like `measurementIndex` would let the reader more easily see `millis` is indexed by measurement. A new name for `millis` might help (`measurementResults`?), `numPending` could be `pendingMeasurementCount`, etc. You could see if you can address this in one of the next PRs too.

File: src/automatic.ts (hunk 1 above)
Comment (aomarks) [reply src/automatic.ts]: Done.

File: src/automatic.ts (hunk 2 above)
Comment (aomarks) [reply src/automatic.ts]: Done, renamed a bunch of stuff, removed a few variables. Still could probably be refactored to be more clear, but hopefully this is a good enough improvement for now.
